### ISSUE TITLE
`attention_mask` padded with 1

### ISSUE THREAD

**Issue by signalogy** [JoakimEdin/medical-coding-reproducibility#13]
Hi, thanks for sharing :-) I find this very helpful. 

A question I have is regarding how you pad the `attention_mask` during collage data. 

https://github.com/JoakimEdin/medical-coding-reproducibility/blob/7b4ae69c1aeeda5653cf54ad9714319376a72797/src/data/datasets.py#L76-L78

Since `seq2batch` function pads with `self.tokenizer.pad_token_id`, the `attention_mask` variable is padded with `1`. Shouldn't it be padded with `0` as padded tokens should not be used in attention?

Thanks,

**Comment by JoakimEdin** [comment]
You are correct! That is an error. 

I don't think it will make a big difference performance-wise, but it's definitely wrong!

**Comment by JoakimEdin** [comment]
I will not change the code because I want people to be able to reproduce the results from my paper. I will write a warning in the README

**Comment by signalogy** [comment]
Thanks for the prompt response. 

I re-ran the experiment with fixed `attention_mask` filled with 0 for padded tokens (MIMIC4-ICD10 with PLM-ICD method). The results seem notably lower than the original implementation as shown in the table. 

![image](https://github.com/JoakimEdin/medical-coding-reproducibility/assets/28071380/43a31e1d-87fa-45a3-bcf5-f2177fd100c8)

It was interesting to see the erroneous method outperforms. So, I've checked the self-attention output from RoBERTa model during inference; I found the padded tokens' output nearly zero with the original implementation, while zone-zero values are concentrated at the beginning of the sentence as shown in the figure that depicts self-attention output during inference of a sentence whose length is 100 with 28 padded tokens (left) `attention_mask` is padded with 1 for padded tokens (original implementation) (right) `attention_mask` is padded with 0 for padded tokens.

![image](https://github.com/JoakimEdin/medical-coding-reproducibility/assets/28071380/a380baef-8de6-4283-8027-e70de4874c2a)

Do you have any insights to share regarding this?

**Comment by JoakimEdin** [comment]
Interesting! I would double-check if the original model is better because of the random seed. 

Perhaps RoBERTa uses the PAD tokens for compute and to store information?

Which layer are you looking at?
